extensions/ql-vscode/src/view/results.tsx

```diff
@@ -174,7 +176,10 @@ class App extends React.Component<{}, ResultsViewState> {
     resultsInfo: ResultsInfo
   ): Promise<readonly ResultSet[]> {
     const parsedResultSets = resultsInfo.parsedResultSets;
-    return [{ t: 'RawResultSet', ...parsedResultSets.resultSet }];
+    return [{
+      ...parsedResultSets.resultSet,
+      t: (parsedResultSets.resultSet.t ?? 'RawResultSet') as any
```

Comment:
forgot that null-coalescing actually made it into the language!

can you leave a comment that explains the situation in which `parsedResultSets.resultSet.t` can either be or not be undefined, though? I'm actually confused why the types allow that. If this could be rephrased to avoid the `as any` that would be even better, but I'm not religiously opposed to it if it makes sense.

Reply:
I honestly can't remember. I'll check on this.

Reply:
Right...this is was always the behaviour. (ie- use `RawResultSet` if no `t` is specifed, otherwise use whatever is in `t`. The reason why I changed is that the latest version of tsc started screaming about this.

I honestly don't know which situations we'll have an empty `t` and if we do, it's probably a bug. I'm partially thinking that we delete it and log any errors.

Reply:
Hm, ok, then I consider the way the code is now to be a sufficient red flag that this should ideally be fixed. Thanks, tsc :+1: